src/message.rs

```diff
@@ -611,6 +614,36 @@ impl Message {
         .await
     }
 
+    // It's a little unfortunate that the UI has to first call dc_msg_get_override_sender_name() and then if it was NULL, call
+    // dc_contact_get_display_name() but this was the best solution:
+    // - We could load a Contact struct from the db here to call get_display_name() instead of returning None, but then we had a db
+    //   call everytime (and this fn is called a lot while the user is scrolling through a group), so performance would be bad
```

Comment:
We do these calls all the time for normal groups and performance is ok, no? If performance is the same as for groups, I'd really prefer to have simpler API not exposing implementation details, especially if there are no measurements showing bad performance. Even on HDD, contacts should be cached by the filesystem cache.

I load contact every time in a dumb way in KDeltaChat, and have seen no performance issues on HDD.

Reply:
We do not call these for normal groups, no. Currently the UI loads the contact once for a message and then gets the name. With the simpler API, this function will have to load it a second time in groups (btw performance would go down for groups, not for mailing lists as for mailing lists we have OverrideDisplayname set and don't need to load the name).

But yes, I didn't measure and have no idea by what amount performance would go down.

Reply:
i'd say, the maybe a bit more complicated `dc_msg_get_override_sender_name()` is reasonable here, . it also has the advantage that it can be used to checked if the email-address is actually impersonated (the  email-address  is used by several people). maybe we want to show an icon for that at some point (iirc, there were some requests in this direction wrt bots, not sure, if this matches the current api completely, though)